### PR TITLE
#7 Update and delete user through fosrestbundle api

### PR DESCRIPTION
**Pull request by murugadasvps**

### CODE REVIEW

File: UserApiProject/src/Controller/UserController.php [hunk 1]
```diff
@@ -39,18 +40,18 @@ public function newUser(Request $request)
...
-            ->add('userSchooling', TextType::class)
-            ->add('userUgDegree', TextType::class)
-            ->add('userPgDegree', TextType::class)
-            ->add('userMasterDegree', TextType::class)
-            ->add('userEducationOthers', TextType::class)
-            ->add('userBloodGroup', TextType::class)
-            ->add('userGender', TextType::class)
+            ->add('userFirstName', TextType::class, array("label"=>'First Name'))
+            ->add('userLastName', TextType::class, array("label"=>'Last Name'))
+            ->add('userEmailIds', TextType::class, array("label"=>'Email Ids'))
```
Comment (aspire-shatheesh) [inline UserApiProject/src/Controller/UserController.php]: how to manage multiple email ids? will this support multiple email ids?

File: UserApiProject/src/Controller/UserController.php [hunk 2]
```diff
@@ -39,18 +40,18 @@ public function newUser(Request $request)
...
+            ->add('userPhone', TextType::class, array("label"=>'Mobile/phone Nos'))
```
Comment (aspire-shatheesh) [inline UserApiProject/src/Controller/UserController.php]: same with phone numbers

File: UserApiProject/src/Controller/UserController.php [hunk 3]
```diff
@@ -39,18 +40,18 @@ public function newUser(Request $request)
 
          $form = $this->createFormBuilder($user)
```
Comment (aspire-shatheesh) [inline UserApiProject/src/Controller/UserController.php]: also which is the suggested approach in symfony for forms?

File: UserApiProject/src/Controller/UserController.php [hunk 4]
```diff
@@ -119,7 +150,63 @@ public function postUserAction(Request $request)
...
+     * @Method("POST")
+     */
+    public function updateUserAction(Request $request) 
+    {
+        $dm = $this->get('doctrine_mongodb')->getManager();
+
+        $form_values = $request->request->get('form');
+        $user = $dm->getRepository(userDetails::class)->find($form_values['id']);
+
+        $userDOb = $form_values['userDob']['year'].'-'.$form_values['userDob']['month'].'-'.$form_values['userDob']['day'];
```
Comment (aspire-shatheesh) [inline UserApiProject/src/Controller/UserController.php]: is there any symfony suggested approach for  manipulating form values before save?

File: UserApiProject/src/Controller/UserController.php (hunk 1 above)
Comment (murugadasvps) [reply UserApiProject/src/Controller/UserController.php]: In the current flow, multiple emailids and phone nos can be handled with comma separated in a single text box. And I coverted it into array through explode function and revert to string with implode function. But the absolute solution should be adding each in a separate textbox and make it as an array on direct submit request. Will try and work on the direct request of Symfony forms.

File: UserApiProject/src/Controller/UserController.php (hunk 3 above)
Comment (murugadasvps) [reply UserApiProject/src/Controller/UserController.php]: The suggested method is the form elements should get populated directly on fetching the record and assign it to the createform builder. As started working on symfony, went with this method. Will fine tune my code and the form method request and submit actions.

File: UserApiProject/src/Controller/UserController.php (hunk 1 above)
Comment (aspire-shatheesh) [reply UserApiProject/src/Controller/UserController.php]: Will it violate DB design? What is the DB rules that we should take care in these scenarios?

Comment (murugadasvps) [reply UserApiProject/src/Controller/UserController.php]: It should not violate DB design. As the data type in table is array for email and phone, We have to save the values in array format.

Comment (aspire-shatheesh) [reply UserApiProject/src/Controller/UserController.php]: how to store array in mysql. 
Which data type you are using and 
how the data will be stored?

Comment (murugadasvps) [reply UserApiProject/src/Controller/UserController.php]: In mysql, we can set data type as long text and make use of functions such as serialize and unserialize to store an array. Or else we can go for a separate table and map the parent id on that table.

Comment (murugadasvps) [reply UserApiProject/src/Controller/UserController.php]: Regarding DB rules, for saving as array one-to-many should be achievable. And the storing values should not be in a format other than array.

Comment (aspire-shatheesh) [reply UserApiProject/src/Controller/UserController.php]: Can you repeat. Can you explain further.

Comment (murugadasvps) [reply UserApiProject/src/Controller/UserController.php]: I have created as array type for saving multiple emailids as we can save values as array in that. It falls under one to many as a user can have multiple emailids. Hence we have to pass values in array format to achieve this. The same is applicable for phone nos.

Comment (aspire-shatheesh) [reply UserApiProject/src/Controller/UserController.php]: Okay. What will be the data type used or how it.will be stored in DB

Comment (murugadasvps) [reply UserApiProject/src/Controller/UserController.php]: In code I specified the field type as hash and declare the variable name as array(). In Mongodb it internally save the values as object type which consists of string array.

**Pull request merged**